Steer/DigitizerWorkflow/src/CTPDigitizerSpec.cxx

```diff
@@ -0,0 +1,105 @@
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "CTPDigitizerSpec.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework//Task.h"
+#include "DetectorsBase/BaseDPLDigitizer.h"
+#include "DataFormatsCTP/Digits.h"
+#include "Steer/HitProcessingManager.h" // for DigitizationContext
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "CTPSimulation/Digitizer.h"
+#include "DataFormatsFT0/Digit.h"
+#include "DataFormatsFV0/BCData.h"
+
+#include <TStopwatch.h>
+#include <gsl/span>
+
+using namespace o2::framework;
+namespace o2
+{
+namespace ctp
+{
+class CTPDPLDigitizerTask : public o2::base::BaseDPLDigitizer
+{
+  using GRP = o2::parameters::GRPObject;
+
+ public:
+  CTPDPLDigitizerTask() : o2::base::BaseDPLDigitizer(), mDigitizer() {}
+  ~CTPDPLDigitizerTask() override = default;
+  void initDigitizerTask(framework::InitContext& ic) override
+  {
+    mDigitizer.init();
+  }
+  void run(framework::ProcessingContext& pc)
+  {
+    // read collision context from input
+    //auto context = pc.inputs().get<o2::steer::DigitizationContext*>("collisioncontext");
+    //const bool withQED = context->isQEDProvided();
+    //auto& timesview = context->getEventRecords(withQED);
+    // read ctp inputs from input
+    auto ft0inputs = pc.inputs().get<gsl::span<o2::ft0::DetTrigInput>>("ft0");
+    auto fv0inputs = pc.inputs().get<gsl::span<o2::fv0::DetTrigInput>>("fv0");
+
+    // if there is nothing to do ... return
+    if ((ft0inputs.size() == 0) && (fv0inputs.size() == 0)) {
+      return;
+    }
+    std::vector<o2::ctp::CTPInputDigit> finputs;
+    TStopwatch timer;
+    timer.Start();
+    LOG(INFO) << "CALLING CTP DIGITIZATION";
+    for (const auto& inp : ft0inputs) {
+      CTPInputDigit finpdigit(inp.mIntRecord, inp.mInputs, o2::detectors::DetID::FT0);
+      finputs.emplace_back(finpdigit);
+    }
+    for (const auto& inp : fv0inputs) {
+      CTPInputDigit finpdigit(inp.mIntRecord, inp.mInputs, o2::detectors::DetID::FT0);
+      finputs.emplace_back(finpdigit);
+    }
+    gsl::span<CTPInputDigit> ginputs(finputs);
+    mDigitizer.process(ginputs, mDigits);
+    // send out to next stage
+    LOG(INFO) << "CTP DIGITS being sent.";
+    pc.outputs().snapshot(Output{"CTP", "DIGITS", 0, Lifetime::Timeframe}, mDigits);
+    LOG(INFO) << "CTP PRESENT being sent.";
+    pc.outputs().snapshot(Output{"CTP", "ROMode", 0, Lifetime::Timeframe}, mROMode);
+    timer.Stop();
+    LOG(INFO) << "CTP Digitization took " << timer.CpuTime() << "s";
+  }
+
+ protected:
+  o2::parameters::GRPObject::ROMode mROMode = o2::parameters::GRPObject::PRESENT;
+  gsl::span<o2::ctp::CTPRawData> mDigits;
```

Comment:
See my comment on Digitizer::process method signature. You cannot assign to this span the way you did it and in any case, you should not assign something which exists only in this process method.

Reply:
Right, so span is like pointer. But this is solved by your next suggestion by removing mDigits at all.